pkg/scheduler/framework/plugins/podtopologyspread/plugin.go

```diff
@@ -257,23 +254,43 @@ func (pl *PodTopologySpread) isSchedulableAfterNodeChange(logger klog.Logger, po
 		return framework.Queue, err
 	}
 
+	// We return Queue when either of the following is met -
+	// 1. Node/Delete: the original node had a topology key that matches the pod's topology spread constraints.
+	// 2. Node/UpdateNodeLabels: the original node did not match the constraints but the modified node does.
```

Comment:
OK, similar to [my notice in the podaffinity PR](https://github.com/kubernetes/kubernetes/pull/127470/files#r1770581857), we should return Queue both when non-match -> match and when match -> non-match.

1. Pods are spreading 1:2:2 and they have topologyspread with `maxSkew:1`. So, a new Pod can only go to the first domain.
2. If one node in the third domain gets an update in its labels, and no longer belongs to the third domain, the situation is gonna be 1:2:1, and a new Pod can go to the third domain too.

Reply:
fixed.

Reply:
```go
                // framework.Update: Return Queue if the original node did not match the constraints but the modified node does.
		if !nodeLabelsMatchSpreadConstraints(originalNode.Labels, constraints) && nodeLabelsMatchSpreadConstraints(modifiedNode.Labels, constraints) {
			logger.V(5).Info("the node is updated and now matches all pod topology spread constraints",
				"pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
			return framework.Queue, nil
		}

		// framework.Update: Return Queue if the original node matched the constraints but the modified node does not.
		if nodeLabelsMatchSpreadConstraints(originalNode.Labels, constraints) && !nodeLabelsMatchSpreadConstraints(modifiedNode.Labels, constraints) {
			logger.V(5).Info("the node is updated and no longer matches all pod topology spread constraints",
				"pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
			return framework.Queue, nil
		}
```

Reply:
`nodeLabelsMatchSpreadConstraints` indicates whether the node labels all satisfy the TopologyKey in Constraints. Do we still need to implement match -> not match and not match -> match? If needed, https://github.com/kubernetes/kubernetes/pull/127478#discussion_r1778345551 According to this comment, pod3 and pod4 should be added to the queue at the same time. This is because fake-node2 has a node label (match pod SpreadConstraint) but changes to a zone label (not match SpreadConstraint). @sanposhiho